src/Views/UserSettingsView.vala

```diff
@@ -323,12 +324,23 @@ namespace SwitchboardPlugUserAccounts.Widgets {
                 update_account_type ();
             }
 
-            if (delta_user.automatic_login != user.get_automatic_login ()) {
-                update_autologin ();
+            var user_automatic_login = user.get_automatic_login ();
+            if (delta_user.automatic_login != user_automatic_login) {
+                if (user_automatic_login && !autologin_switch.active) {
+                    autologin_switch.active = true;
+                } else if (!user_automatic_login && autologin_switch.active) {
+                    autologin_switch.active = false;
+                }
             }
 
-            if (delta_user.locked != user.get_locked ()) {
-                update_lock_state ();
+            if (delta_user.locked != user_locked) {
+                if (user_locked) {
+                    enable_user_button.label = _("Enable User Account");
+                    enable_user_button.get_style_context ().remove_class (Gtk.STYLE_CLASS_DESTRUCTIVE_ACTION);
+                } else {
+                    enable_user_button.label = _("Disable User Account");
+                    enable_user_button.get_style_context ().add_class (Gtk.STYLE_CLASS_DESTRUCTIVE_ACTION);
+                }
```

Comment:
The condition `delta_user.locked != user_locked` seems to prevent `enable_user_button` from setting its label on the first load:

![Screenshot from 2019-10-24 19-25-05](https://user-images.githubusercontent.com/26003928/67477063-46819180-f694-11e9-9978-b284e55985f6.png)

Reply:
Oh yeah that's no good. Fixed!